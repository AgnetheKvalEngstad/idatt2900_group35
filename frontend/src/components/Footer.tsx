import React from "react";
import { Grid2, Button } from "@mui/material";
import QuestionMarkIcon from "@mui/icons-material/QuestionMark";
import TextSizeDialog from "./dialogs/TextSizeDialog";
import HelpDialog from "./dialogs/HelpDialog";
import { useNavigate } from "react-router-dom";
import { useLocation } from "react-router-dom";

/**
 * A React component that renders a footer with buttons for increasing text size
 * and getting information and instructions for the page.
 *
 * @returns A React component with buttons for increasing text size and accessing help.
 */
export default function Footer() {
  const [openTextSize, setOpenTextSize] = React.useState(false);
  const [openHelp, setOpenHelp] = React.useState(false);
  const location = useLocation();
  const navigate = useNavigate();

  const handleTextSizeOpen = () => setOpenTextSize(true);
  const handleTextSizeClose = () => setOpenTextSize(false);

  const handleHelpOpen = () => setOpenHelp(true);
  const handleHelpClose = () => setOpenHelp(false);

  return (
    <Grid2 className="sticky bottom-0 mt-auto flex flex-row items-center justify-between pl-4 py-2 bg-white">
      <Grid2 className="flex flex-col gap-2">
        <Button
          variant="contained"
          color="primary"
          onClick={handleTextSizeOpen}
        >
          Stor skrift
        </Button>
        <TextSizeDialog open={openTextSize} onClose={handleTextSizeClose} />

        <Button
          variant="contained"
          color="primary"
          startIcon={<QuestionMarkIcon />}
          onClick={handleHelpOpen}
        >
          Hjelp
        </Button>
        <HelpDialog open={openHelp} onClose={handleHelpClose} />
      </Grid2>

      {location.pathname === "/" && (
        <Grid2 className="pr-8">
          <Button
            variant="contained"
            color="primary"
            onClick={() => navigate("/home")}
          >
            Hopp over
          </Button>
        </Grid2>
      )}
    </Grid2>
  );
}
