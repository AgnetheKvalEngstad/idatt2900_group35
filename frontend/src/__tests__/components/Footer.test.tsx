import { render, screen } from "@testing-library/react";
import Footer from "../../components/Footer";
import "@testing-library/jest-dom";

/**
 * Tests for Footer component
 */
describe("Tests for component Footer", () => {
  beforeEach(() => {
    render(<Footer />);
  });

  it("should render Footer with a button for larger text", () => {
    const button = screen.getByText("Stor skrift");
    expect(button).toBeInTheDocument();
  });

  it("should render Footer with a button for help", () => {
    const button = screen.getByText("Hjelp");
    expect(button).toBeInTheDocument();
  });
});
