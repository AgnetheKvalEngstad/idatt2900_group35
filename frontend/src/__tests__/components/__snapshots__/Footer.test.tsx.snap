// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests for component Footer > should render Footer 1`] = `
<ForwardRef(Grid)
  className="mt-auto flex pl-4 py-2"
  container={true}
  direction="column"
  spacing={2}
>
  <ForwardRef(Grid)>
    <ForwardRef(Button)
      color="primary"
      variant="contained"
    >
      Stor skrift
    </ForwardRef(Button)>
  </ForwardRef(Grid)>
  <ForwardRef(Grid)>
    <ForwardRef(Button)
      color="primary"
      startIcon={<Memo />}
      variant="contained"
    >
      Hjelp
    </ForwardRef(Button)>
  </ForwardRef(Grid)>
</ForwardRef(Grid)>
`;
